VoiceController.php

```diff
@@ -4,45 +4,33 @@ public function voice(Request $request){
         'value'=>'required|boolean',
     ]);
 
-    $question=Question::find($request->post('question_id'));
-    if (!$question)
-        return response()->json([
-            'status'=>404,
-            'message'=>'not found question ..'
-        ]);
-    if ($question->user_id==auth()->id())
-        return response()->json([
-            'status' => 500,
-            'message' => 'The user is not allowed to vote to your question'
-        ]);
+    $question = Question::find($request->question_id);
 
-    //check if user voted 
-    $voice=Voice::where([
-        ['user_id','=',auth()->id()],
-        ['question_id','=',$request->post('question_id')]
-    ])->first();
-    if (!is_null($voice)&&$voice->value===$request->post('value')) {
-        return response()->json([
-            'status' => 500,
-            'message' => 'The user is not allowed to vote more than once'
-        ]);
-    }else if (!is_null($voice)&&$voice->value!==$request->post('value')){
-        $voice->update([
-            'value'=>$request->post('value')
-        ]);
-        return response()->json([
-            'status'=>201,
-            'message'=>'update your voice'
-        ]);
+    // We dont have to check if Question not exists. Because validation checks for us
```

Comment:
Nice catch with validation that question exists. I think you're the only one who payed attention to that.

Reply:
Sometimes my coworkers don't use validations at all. I don't know why but for some reason they don't use all the features in framework.